src/poetry/core/semver/version_range.py

```diff
@@ -69,27 +52,35 @@ def is_simple(self) -> bool:
 
     def allows(self, other: Version) -> bool:
         if self._min is not None:
-            if other < self._min:
+            _this, _other = self.allowed_min, other
+
+            assert _this is not None
+
+            if not _this.is_postrelease() and _other.is_postrelease():
+                _other = _other.without_postrelease()
```

Comment:
Seems unintuitive at first glance. Maybe, we should add a comment like `# required for "2.0.post1" does not match ">2"`